src/MyEventEmitter.js

```diff
@@ -1,14 +1,68 @@
+/* eslint-disable max-len */
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.events = {};
+  }
+
+  checkEventName = (eventName) => {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+  };
+
+  on(eventName, listener) {
+    this.checkEventName(eventName);
+    this.events[eventName].push(listener);
+  }
+
+  once(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.on(eventName, onceListener);
+  }
+
+  off(eventName, listener) {
+    this.events[eventName] = this.events[eventName].filter(
+      (registeredListener) => registeredListener !== listener,
+    );
+  }
+
+  emit(eventName, ...args) {
+    this.events[eventName].forEach((listener) => {
+      listener(...args);
+    });
+  }
+
+  prependListener(eventName, listener) {
+    this.checkEventName(eventName);
+    this.events[eventName].unshift(listener);
+  }
+
+  prependOnceListener(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.prependListener(eventName, onceListener);
+  }
+
+  removeAllListeners(eventName) {
+    if (eventName) {
+      delete this.events[eventName];
+    } else {
+      this.events = {};
+    }
+  }
+
+  listenerCount(eventName) {
+    return this.events[eventName] ? this.events[eventName].length : 0;
```

Comment:
The `listenerCount` method doesn't check if the event exists before trying to get its listener count. If the event doesn't exist, an error will be thrown. To fix this, you should add a check for the event's existence before trying to get its listener count.